t/unit-tests/t-reftable-pq.c

```diff
@@ -45,9 +50,11 @@ static void test_pq_record(void)
 	} while (i != 1);
 
 	while (!merged_iter_pqueue_is_empty(pq)) {
+		struct pq_entry top = merged_iter_pqueue_top(pq);
 		struct pq_entry e = merged_iter_pqueue_remove(&pq);
 		merged_iter_pqueue_check(&pq);
 
+		check(pq_equal(&top, &e));
```

Comment:
I think it would've made sense to put this into a separate test case as now you do explicitly verify a diferent part of the codebase. Also, I think it would be preferable to not compare the results of those two functions, but instead to directly verify our assumptions.

Reply:
>Also, I think it would be preferable to not compare the results of those two functions, but instead to directly verify our assumptions.

I'm not sure I follow what you mean by that. `merged_iter_pqueue_top()` returns the same value as the next call to `merged_iter_pqueue_remove()` would, so wouldn't it be natural to verify the correctness of `merged_iter_pqueue_top()` by comparing the output of both these function? The validity of the priority queue is checked with `merged_iter_pqueue_check()` after each call to `merged_iter_pqueue_remove()`, which validates `merged_iter_pqueue_remove()` and hence, `merged_iter_pqueue_top()` as well.

Reply:
I at least wouldn't do this exclusively. It's feasible that both functions have a similar bug, or that the latter is built on top of the former. So it feels a bit like one might be lifting oneself up by one's bootstraps.

In any case, I don't think it hurts to compare the results of those functions. I just think it would be nice to also have an explicit check that our assumptions hold true.